### PR TITLE
HOCS-3239 Adding and testing management of MPAM business units

### PR DESCRIPTION
**Pull request by gardnersam**

### CODE REVIEW

File: src/shared/pages/list/mpamBusinessUnits/__tests__/__snapshots__/businessAreaView.spec.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`when the businessAreaView component is mounted should render with default props 1`] = `<div />`;
```
Comment (robertdeniszczyc2) [inline src/shared/pages/list/mpamBusinessUnits/__tests__/__snapshots__/businessAreaView.spec.tsx.snap]: This looks suspiciously empty, worth double checking this test is performing as anticipated/required?

File: src/shared/pages/list/mpamBusinessUnits/__tests__/addBusinessUnit.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
+
```
Comment (robertdeniszczyc2) [inline src/shared/pages/list/mpamBusinessUnits/__tests__/addBusinessUnit.spec.tsx]:
```suggestion
```

File: src/shared/pages/list/mpamBusinessUnits/__tests__/amendBusinessUnit.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
+
```
Comment (robertdeniszczyc2) [inline src/shared/pages/list/mpamBusinessUnits/__tests__/amendBusinessUnit.spec.tsx]:
```suggestion
```

File: src/shared/pages/list/mpamBusinessUnits/addBusinessUnit.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                    />
+                    <h1 className="govuk-heading-xl">
+                        Add Business Unit to {readableType}
+                    </h1>
+                </div>
+            </div>
+            <div className="govuk-grid-row">
+                <div className="govuk-grid-column-one-half-from-desktop">
+                    <form action={`/api/entity/list/${type}`} method="POST" onSubmit={handleSubmit}>
+                        <Text
```
Comment (robertdeniszczyc2) [inline src/shared/pages/list/mpamBusinessUnits/addBusinessUnit.tsx:78]: Should there be a CSRF token here as there is on amendBusinessUnit.tsx?

File: src/shared/pages/list/mpamBusinessUnits/amendBusinessUnit.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
+import React, { Reducer, useEffect } from 'react';
```
Comment (robertdeniszczyc2) [inline src/shared/pages/list/mpamBusinessUnits/amendBusinessUnit.tsx:1]: File name is amendBusinessUnit but inside the file it's businessArea, is this intentional?

File: src/shared/pages/list/mpamBusinessUnits/listNames.tsx [hunk 6]
```diff
@@ -0,0 +1,8 @@
+export const listNames = {
+    MPAM_BUS_UNITS_1UKVI: 'UKVI',
```
Comment (robertdeniszczyc2) [inline src/shared/pages/list/mpamBusinessUnits/listNames.tsx]: This constant doesn't match the convention of the others, should it be sans UKVI?

File: src/shared/pages/list/mpamBusinessUnits/listNames.tsx [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+    MPAM_BUS_UNITS_2: 'BF',
+    MPAM_BUS_UNITS_3: 'IE',
+    MPAM_BUS_UNITS_4: 'EUSS',
+    MPAM_BUS_UNITS_5: 'HMPO',
+    MPAM_BUS_UNITS_6: 'Windrush',
+    MPAM_BUS_UNITS_7: 'Corona Virus' };
```
Comment (robertdeniszczyc2) [inline src/shared/pages/list/mpamBusinessUnits/listNames.tsx]: Is this the correct stylisation of coronavirus/corona virus?

File: src/shared/pages/list/mpamBusinessUnits/selectBusinessArea.tsx [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    const [selectedCaseType, setSelectedCaseType] = React.useState<Item>();
+
+    const items: Item[] = [
+        { label: 'UKVI', value: 'MPAM_BUS_UNITS_1' },
+        { label: 'BF', value: 'MPAM_BUS_UNITS_2' },
+        { label: 'IE', value: 'MPAM_BUS_UNITS_3' },
+        { label: 'EUSS', value: 'MPAM_BUS_UNITS_4' },
+        { label: 'HMPO', value: 'MPAM_BUS_UNITS_5' },
+        { label: 'Windrush', value: 'MPAM_BUS_UNITS_6' },
+        { label: 'Corona Virus', value: 'MPAM_BUS_UNITS_7' },
```
Comment (robertdeniszczyc2) [inline src/shared/pages/list/mpamBusinessUnits/selectBusinessArea.tsx]: Duplication of strings for the labels here from listNames.tsx, potential to use the constant values?

**Review by robertdeniszczyc2 — COMMENTED** [review]
Added a few comments/questions, any questions give me a shout,

Cheers,
Rob

File: src/shared/pages/list/mpamBusinessUnits/listNames.tsx (hunk 6 above)
Comment (gardnersam) [reply src/shared/pages/list/mpamBusinessUnits/listNames.tsx]: Yes I'll remove the UKVI

File: src/shared/pages/list/mpamBusinessUnits/listNames.tsx (hunk 7 above)
Comment (gardnersam) [reply src/shared/pages/list/mpamBusinessUnits/listNames.tsx]: No, I'll update to match elsewhere (Coronavirus (COVID-19))

File: src/shared/pages/list/mpamBusinessUnits/amendBusinessUnit.tsx (hunk 5 above)
Comment (gardnersam) [reply src/shared/pages/list/mpamBusinessUnits/amendBusinessUnit.tsx:1]: No it should all be about amending units, I'll change it

File: src/shared/pages/list/mpamBusinessUnits/selectBusinessArea.tsx (hunk 8 above)
Comment (gardnersam) [reply src/shared/pages/list/mpamBusinessUnits/selectBusinessArea.tsx]: swapped to generate this array from listNames.tsx

File: src/shared/pages/list/mpamBusinessUnits/__tests__/__snapshots__/businessAreaView.spec.tsx.snap (hunk 1 above)
Comment (gardnersam) [reply src/shared/pages/list/mpamBusinessUnits/__tests__/__snapshots__/businessAreaView.spec.tsx.snap]: Fixed, nice and full now

**Review by robertdeniszczyc2 — APPROVED** [review]

**Pull request merged**